truss/contexts/image_builder/cache_warmer.py

```diff
@@ -48,106 +66,141 @@ def split_path(path, prefix="gs://"):
     return bucket_name, path
 
 
-def parse_s3_service_account_file(file_path):
-    # open the json file
-    with open(file_path, "r") as f:
-        data = json.load(f)
+class RepositoryFile:
+    def __init__(self, repo_name, file_name, revision_name):
+        self.repo_name = repo_name
+        self.file_name = file_name
+        self.revision_name = revision_name
+        self.client = None
+        self.bucket_name = None
+        self.bucket = None
+        self.is_private = True
+
+    @staticmethod
+    def create(repo_name, file_name, revision_name):
+        if repo_name.startswith("gs://"):
+            repository_class = GCSFile
+        elif repo_name.startswith("s3://"):
+            repository_class = S3File
+        else:
+            repository_class = HuggingFaceFile
+        return repository_class(repo_name, file_name, revision_name)
+
+    def connect(self, key_file=None):
+        # connect to the correct repo and set up accessors
+        pass
+
+    def download_to_cache(self):
+        pass
+
+
+class HuggingFaceFile(RepositoryFile):
+    def download_to_cache(self):
+        secret_path = Path("/etc/secrets/hf-access-token")
+        secret = secret_path.read_text().strip() if secret_path.exists() else None
+        try:
+            hf_hub_download(
+                self.repo_name,
+                self.file_name,
+                revision=self.revision_name,
+                token=secret,
+            )
+        except FileNotFoundError:
+            raise RuntimeError(
+                "Hugging Face repository not found (and no valid secret found for possibly private repository)."
+            )
 
-    # validate the data
-    if "aws_access_key_id" not in data or "aws_secret_access_key" not in data:
-        raise ValueError("Invalid AWS credentials file")
 
-    # parse the data
-    aws_access_key_id = data["aws_access_key_id"]
-    aws_secret_access_key = data["aws_secret_access_key"]
-    aws_region = data["aws_region"]
+class GCSFile(RepositoryFile):
+    def connect(self, key_file="/app/data/service_account.json"):
```

Comment:
what is the point of this `connect` function? could we do all of this as a part of `download`?